### PR TITLE
Update kubebuilder V4

### PR DESCRIPTION
**Pull request by MarcosDY**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -121,10 +127,11 @@ lint-code: $(golangci_lint_bin)
 
 ##@ Build
 
+# TODO: VERIFY BUILD KEEP WORKING WIHT NEW MAIN!!!!!!!!!!!!!!!
```
Comment (azdagron) [inline Makefile:130]: Did the build keep working 😄 ?

File: cmd/main.go [hunk 2]
```diff
@@ -93,6 +92,21 @@ func parseConfig() (spirev1alpha1.ControllerManagerConfig, ctrl.Options, []*rege
 			"Command-line flags override configuration from this file.")
 	flag.StringVar(&spireAPISocketFlag, "spire-api-socket", "", "The path to the SPIRE API socket (deprecated; use the config file)")
 
+	// TODO: may we get metric, probeAdrr and leader?
```
Comment (azdagron) [inline cmd/main.go:95]: Has this commented code been addressed elsewhere? the features provided by these configurables are still necessary, especially leader election....

File: cmd/main.go [hunk 3]
```diff
@@ -274,41 +288,47 @@ func run(ctrlConfig spirev1alpha1.ControllerManagerConfig, options ctrl.Options,
...
 		Scheme:    mgr.GetScheme(),
 		Triggerer: entryReconciler,
 	}).SetupWithManager(mgr); err != nil {
 		setupLog.Error(err, "unable to create controller", "controller", "ClusterStaticEntry")
 		return err
 	}
-	if err = (&spirev1alpha1.ClusterFederatedTrustDomain{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "webhook", "ClusterFederatedTrustDomain")
-		return err
+	if os.Getenv("ENABLE_WEBHOOKS") != "false" {
```
Comment (azdagron) [inline cmd/main.go:317]: @kfox1111, you might be interested in this. Looks like a new kubebuilder convention?

File: config/crd/bases/spire.spiffe.io_clusterfederatedtrustdomains.yaml [hunk 4]
```diff
@@ -88,9 +87,3 @@ spec:
     storage: true
     subresources:
       status: {}
-status:
```
Comment (azdagron) [inline config/crd/bases/spire.spiffe.io_clusterfederatedtrustdomains.yaml:91]: I believe we still need these. For some reason kubebuilder isn't generating them but without them the status subresource does not update. See #223.

File: demo/test.sh [hunk 5]
```diff
@@ -37,7 +37,7 @@ cleanup() {
     echo "Done."
 }
 
-trap cleanup EXIT
+# trap cleanup EXIT
```
Comment (azdagron) [inline demo/test.sh:40]: Was this commented out for testing?

**Review by azdagron — COMMENTED** [review]
Did you manually verify things worked with this change? In particular, I believe this change as currently written will break the updating of the status subresource on the CRs...

File: Makefile (hunk 1 above)
Comment (MarcosDY) [reply Makefile:130]: hahah forgot to remove the reminders haha

File: demo/test.sh (hunk 5 above)
Comment (MarcosDY) [reply demo/test.sh:40]: yeap

File: config/crd/bases/spire.spiffe.io_clusterfederatedtrustdomains.yaml (hunk 4 above)
Comment (MarcosDY) [reply config/crd/bases/spire.spiffe.io_clusterfederatedtrustdomains.yaml:91]: code generator is no longer [creating that status](https://github.com/kubernetes-sigs/controller-tools/pull/630),
but when I run demo without status I got no error, and status is generated:
```
./cluster1 kubectl describe clusterspiffeid                                                                                                             ✔  16:52:03 
Name:         greeter-server
Namespace:
Labels:       <none>
Annotations:  <none>
API Version:  spire.spiffe.io/v1alpha1
Kind:         ClusterSPIFFEID
Metadata:
  Creation Timestamp:  2023-10-27T19:48:38Z
  Generation:          1
  Resource Version:    725
  UID:                 7043c2fa-90e8-40d8-aa0d-8bd15b0e91eb
Spec:
  Federates With:
    cluster2.demo
  Pod Selector:
    Match Labels:
      spire.spiffe.io/spiffeid:  greeter-server
  Spiffe ID Template:            spiffe://cluster1.demo/greeter-server
Status:
  Stats:
    Entries Masked:             0
    Entries To Set:             1
    Entry Failures:             0
    Namespaces Ignored:         4
    Namespaces Selected:        6
    Pod Entry Render Failures:  0
    Pods Selected:              1
```

not sure, may we add just in case?

File: cmd/main.go (hunk 3 above)
Comment (kfox1111) [reply cmd/main.go:306]: Hadn't seen that. Interesting. Thanks for the heads up.

File: config/crd/bases/spire.spiffe.io_clusterfederatedtrustdomains.yaml (hunk 4 above)
Comment (azdagron) [reply config/crd/bases/spire.spiffe.io_clusterfederatedtrustdomains.yaml]: I see. Was this on a clean cluster? I can check this myself to confirm. Maybe it depends on the version of k8s?

Comment (MarcosDY) [reply config/crd/bases/spire.spiffe.io_clusterfederatedtrustdomains.yaml]: it was in the test cluster, (demo one), but I remember getting the erorr you mentioned in aws... like 3 months ago..
so I think it is safe to keep adding this for now

File: cmd/main.go [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2021 SPIRE Authors.
+Copyright 2023 SPIRE Authors.
```
Comment (azdagron) [inline cmd/main.go:2]: can we revert the copyright year change?

File: internal/controller/clusterfederatedtrustdomain_controller.go [hunk 7]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2021 SPIRE Authors.
```
Comment (azdagron) [inline internal/controller/clusterfederatedtrustdomain_controller.go:2]: Same here. Should revert this change.

File: internal/controller/clusterspiffeid_controller.go [hunk 8]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2021 SPIRE Authors.
+Copyright 2023 SPIRE Authors.
```
Comment (azdagron) [inline internal/controller/clusterspiffeid_controller.go:2]: Same here. Should revert this change.

File: internal/controller/clusterstaticentry_controller.go [hunk 9]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2021 SPIRE Authors.
+Copyright 2023 SPIRE Authors.
```
Comment (azdagron) [inline internal/controller/clusterstaticentry_controller.go:2]: Same here. Should revert this change.

File: internal/controller/common.go [hunk 10]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2021 SPIRE Authors.
+Copyright 2023 SPIRE Authors.
```
Comment (azdagron) [inline internal/controller/common.go:2]: Same here. Should revert this change.

File: internal/controller/pod_controller.go [hunk 11]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2021 SPIRE Authors.
+Copyright 2023 SPIRE Authors.
```
Comment (azdagron) [inline internal/controller/pod_controller.go:2]: Same here. Should revert this change.

**Review by azdagron — COMMENTED** [review]
Just some nits on the copyright year changes, other than that LGTM.

**Review by azdagron — APPROVED** [review]
